m2e-maven-runtime/org.eclipse.m2e.maven.indexer/src/main/java/org/apache/maven/index/NexusArchetypeDataSource.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.apache.maven.index;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.archetype.source.ArchetypeDataSource;
+import org.apache.maven.index.archetype.AbstractArchetypeDataSource;
+import org.apache.maven.index.context.IndexingContext;
+
+/**
+ * Trivial implementation of a Nexus indexer-based data source for archetypes.
+ */
+@Singleton
+@Named("nexus")
+public class NexusArchetypeDataSource extends AbstractArchetypeDataSource implements ArchetypeDataSource {
```

Comment:
Can you please add a not in the class and/or in the commit message about why this class is now necessary?

Reply:
I added some comments to explain why this class exists